src/Streams.jl

```diff
@@ -288,8 +288,7 @@ function Base.readbytes!(http::Stream, buf::IOBuffer, n=bytesavailable(http))
     buf.size += n
 end
 
-function Base.read(http::Stream)
-    buf = PipeBuffer()
+function Base.read(http::Stream, buf::IOBuffer=PipeBuffer())
```

Comment:
If we make buf a `Base.GenericIOBuffer` then we can pass views as well. Same applies fo `Base.readbytes!(http::Stream, buf::IOBuffer)`